src/data/finset.lean

```diff
@@ -2551,6 +2551,44 @@ lemma disjoint_iff_disjoint_coe {α : Type*} {a b : finset α} [decidable_eq α]
   disjoint a b ↔ disjoint (↑a : set α) (↑b : set α) :=
 by { rw [finset.disjoint_left, set.disjoint_left], refl }
 
+/--
+Given a set A and a set B inside it, we can shrink A to any appropriate size, and keep B
+inside it.
+-/
+lemma exists_intermediate_set {A B : finset α} (i : ℕ)
+  (h₁ : card A ≥ i + card B) (h₂ : B ⊆ A) :
+  ∃ (C : finset α), B ⊆ C ∧ C ⊆ A ∧ card C = i + card B :=
+begin
+  rcases nat.le.dest h₁ with ⟨k, _⟩,
+  clear h₁,
+  induction k with k ih generalizing A,
+  { exact ⟨A, h₂, subset.refl _, h.symm⟩ },
+  { have: (A \ B).nonempty,
+    { rw [← card_pos, card_sdiff h₂, ← h, nat.add_right_comm,
+          nat.add_sub_cancel, nat.add_succ],
+      apply nat.succ_pos },
+    rcases this with ⟨a, ha⟩,
+    have z: i + card B + k = card (erase A a),
+    { rw [card_erase_of_mem, ← h, nat.add_succ, nat.pred_succ],
+      rw mem_sdiff at ha,
+      exact ha.1 },
+    rcases ih _ z with ⟨B', hB', B'subA', cards⟩,
+    { exact ⟨B', hB', trans B'subA' (erase_subset _ _), cards⟩ },
+    { rintros t th,
+      apply mem_erase_of_ne_of_mem _ (h₂ th),
+      rintro rfl,
+      rw mem_sdiff at ha,
+      tauto } }
+end
+
+/-- We can shrink A to any smaller size. -/
+lemma exists_smaller_set (A : finset α) (i : ℕ) (h₁ : card A ≥ i) :
+  ∃ (B : finset α), B ⊆ A ∧ card B = i :=
+begin
+  obtain ⟨B, _, x₁, x₂⟩ := exists_intermediate_set i (by simpa) (empty_subset A),
+  exact ⟨B, x₁, x₂⟩,
+end
```

Comment:
Here mathlib style is to compress this trivial proof on one line:
```suggestion
let ⟨B, _, x₁, x₂⟩ := exists_intermediate_set i (by simpa) (empty_subset A) in ⟨B, x₁, x₂⟩
```

Reply:
Ahh I couldn't figure out how to make this a one liner, thanks!